### PR TITLE
Fix printf %p

### PR DESCRIPTION
**Pull request by arichardson**

### CODE REVIEW

File: util/snprintf.c [hunk 1]
```diff
@@ -26,12 +27,12 @@ int vsnprintf(char* out, size_t n, const char* s, va_list vl)
...
         case 'p':
-          longarg = true;
+          uintptrarg = true;
           if (++pos < n) out[pos-1] = '0';
           if (++pos < n) out[pos-1] = 'x';
         case 'x':
         {
-          long num = longarg ? va_arg(vl, long) : va_arg(vl, int);
+          long num = uintptrarg ? va_arg(vl, uintptr_t) : (longarg ? va_arg(vl, long) : va_arg(vl, int));
           for(int i = 2*(longarg ? sizeof(long) : sizeof(int))-1; i >= 0; i--) {
```
Comment (jrtc27) [inline util/snprintf.c]: This will lose the high half of the address currently.

Comment (arichardson) [reply util/snprintf.c]:
```suggestion
          for(int i = 2*((longarg || uintptrarg) ? sizeof(long) : sizeof(int))-1; i >= 0; i--) {
```

**Comment by arichardson** [comment]
ping?

File: machine/mtrap.c [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
 void __attribute__((noreturn)) bad_trap(uintptr_t* regs, uintptr_t dummy, uintptr_t mepc)
 {
-  die("machine mode: unhandlable trap %d @ %p", read_csr(mcause), mepc);
+  die("machine mode: unhandlable trap %ld @ %p", read_csr(mcause), (void*)mepc);
```
Comment (jrtc27) [inline machine/mtrap.c:21]: Is this cast needed?

Comment (arichardson) [reply machine/mtrap.c:21]: Clang warns otherwise, but yeah it is not required unless you want to be c standards compliant.

Comment (jrtc27) [reply machine/mtrap.c:21]: Ugh, you're right, uintptr_t isn't a %p (presumably it's allowed to have a different representation to void*?), you'd have to use PRIxPTR etc instead. So yeah, this is fine...

**Pull request merged**